swarm/storage/localstore.go

```diff
@@ -83,8 +94,19 @@ func (self *LocalStore) CacheCounter() uint64 {
 }
 
 // LocalStore is itself a chunk store
-// unsafe, in that the data is not integrity checked
 func (self *LocalStore) Put(chunk *Chunk) {
+	valid := true
```

Comment:
this crucial code is still incorrect and that it is not caught  by tests is not good. we should test it properly

Reply:
@nolash could you add the tests that cover all combinations of the 2 validators, just so that we lock the behaviour of the different validators kicking-in? I think we discussed this in the call at some point.

It is not clear to me what is wrong with the current code.